core/src/main/java/com/linecorp/armeria/client/retry/RetryRule.java

```diff
@@ -0,0 +1,295 @@
+        return onException(exceptionFilter, Backoff.ofDefault());
+    }
+
+    /**
+     * Returns a newly created {@link RetryRule} that will retry if an {@link Exception} is raised and
+     * the specified {@code exceptionFilter} returns {@code true}.
+     */
+    static RetryRule onException(Predicate<? super Throwable> exceptionFilter, Backoff backoff) {
+        return builder().onException(exceptionFilter).thenBackoff(backoff);
+    }
+
+    /**
+     * Returns a newly created {@link RetryRule} that retries on any {@link Exception}.
+     */
+    static RetryRule onException() {
+        return onException(Backoff.ofDefault());
+    }
+
+    /**
+     * Returns a newly-created {@link RetryRule} that retries with the specified {@link Backoff} on any {@link Exception}.
+     */
+    static RetryRule onException(Backoff backoff) {
+        return builder().onException().thenBackoff(backoff);
+    }
+
+    /**
+     * Returns a {@link RetryRule} that retries on an {@link UnprocessedRequestException} which means that
+     * the request has not been processed by the server. Therefore, you can safely retry the request without
+     * worrying about the idempotency of the request.
+     */
+    static RetryRule onUnprocessed() {
+        return builder().onUnprocessed().thenBackoff();
+    }
+
+    /**
+     * Returns a {@link RetryRule} that retries with the specified {@link Backoff} on
+     * an {@link UnprocessedRequestException} which means that the request has not been processed by the server.
+     * Therefore, you can safely retry the request without worrying about the idempotency of the request.
+     */
+    static RetryRule onUnprocessed(Backoff backoff) {
+        return builder().onUnprocessed().thenBackoff(backoff);
+    }
+
+    /**
+     * Returns composed {@link RetryRule} that represents a logical OR of this {@link RetryRule} and another.
+     * If this {@link RetryRule} completes with {@link RetryRuleDecision#retry(Backoff)} or
+     * {@link RetryRuleDecision#noRetry()}, then other {@link RetryRule} is not evaluated.
+     */
+    default RetryRule or(RetryRule other) {
+        return (ctx, cause) -> {
+            final CompletionStage<RetryRuleDecision> decisionFuture = shouldRetry(ctx, cause);
+            return decisionFuture.thenCompose(decision -> {
+                if (decision != RetryRuleDecision.next()) {
+                    return decisionFuture;
+                } else {
+                    return other.shouldRetry(ctx, cause);
+                }
+            });
+        };
+    }
+
+    /**
+     * Tells whether the request sent with the specified {@link ClientRequestContext} requires a retry or not.
+     * Implement this method to return a {@link CompletionStage} and to complete it with a desired
+     * {@link RetryRuleDecision#retry(Backoff)}.
+     * To stop trying further, complete it with {@link RetryRuleDecision#noRetry()}.
```

Comment:
Instead of `stop trying further` something like `to not retry`. `noRetry` is the opposite of `retry`, not `next` so the wording should be similar to that one